plugin/build_baron.go

```diff
@@ -199,44 +81,93 @@ func BbGetConfig(settings *evergreen.Settings) map[string]evergreen.BuildBaronSe
 	return projects
 }
 
-// BbGetProject retrieves build baron settings from project or admin config depending on PluginAdminPageDisabled flag.
-// Project parser config takes precedence, otherwise fallback to project page settings (found using version ID if given).
+// BbGetProject retrieves build baron settings from project settings.
+// Project page settings takes precedence, otherwise fallback to project config yaml.
 // Returns build baron settings and ok if found.
-func BbGetProject(settings *evergreen.Settings, projectId string, version string) (evergreen.BuildBaronSettings, bool) {
-	flags, err := evergreen.GetServiceFlags()
-	if err != nil {
+func BbGetProject(projectId string, version string) (evergreen.BuildBaronSettings, bool) {
+	projectRef, err := model.FindMergedProjectRef(projectId, version, true)
+	if err != nil || projectRef == nil {
+		return evergreen.BuildBaronSettings{}, false
+	}
+	isValid := validateBbProject(projectId, projectRef.BuildBaronSettings)
+	if !isValid {
 		return evergreen.BuildBaronSettings{}, false
 	}
-	if flags.PluginAdminPageDisabled {
-		if version == "" {
-			lastGoodVersion, err := model.FindVersionByLastKnownGoodConfig(projectId, -1)
-			if err == nil && lastGoodVersion != nil {
-				version = lastGoodVersion.Id
-			}
+	return projectRef.BuildBaronSettings, true
+}
+
+func validateBbProject(projName string, proj evergreen.BuildBaronSettings) bool {
```

Comment:
Do we already do all of this validation in the project page / project patch route? We shouldn't allow users to store an invalid configuration, in which case we shouldn't need to validate here at all.

Reply:
This validation used to happen looping through each project on startup - now that that code is gone it does not exist anywhere but here.  However I do agree it makes sense to move this check into the modifyProject route since that is the only time we are writing to these BB configs.